pkg/client/apifactory_mock.go

```diff
@@ -398,11 +398,21 @@ func (m *MockedAPIProvider) UpdatePriorityClass(oldObj *schedv1.PriorityClass, n
 }
 
 func (m *MockedAPIProvider) GetPodBindStats() BindStats {
-	return m.clients.KubeClient.(*KubeClientMock).GetBindStats()
+	client, ok := m.clients.KubeClient.(*KubeClientMock)
+	if !ok {
+		log.Log(log.Test).Error("failed to get KubeClientMock")
+		return BindStats{}
+	}
```

Comment:
Let this fail immediately, the test must be fixed if this ever happens. Use `nolint` to suppress the linter.

Reply:
Failing immediately would need a panic here which could have side effects and may not allow us to collect all data.
Returning a nil here will cause a failure in the caller which is nicely handled as a failure without the side effects of a panic.

Reply:
Hi @wilfred-s , I checked the usage of `GetPodBindStats`, and it is only referenced in *_test.go. IMHO, hitting a panic in ut files should be acceptable, right?

Reply:
It would be far nicer to just return a nil and let the caller handle it. We finally end up here in `BenchmarkSchedulingThroughPut`:
```
	stat := cluster.GetPodBindStats()
	diff := stat.Last.Sub(stat.First)
	fmt.Printf("Overall throughput: %.0f allocations/s\n", float64(totalPods)/diff.Seconds())
```
Handle the nil there and still print the metrics just after it as the run has finished.

Reply:
I see. I handled the failure and don't print out Overall throughput if stat is nil